07-spinnaker-pipeline-resource/proposal.md

```diff
@@ -0,0 +1,130 @@
+# Summary
+
+Implementing a Spinnaker pipeline resource.  
+
+## Features :
+   - Trigger a Spinnaker pipeline from concourse.
+   - Trigger a concourse job based on the status of a Spinnaker pipeline.
+
+The idea is build on an already existing [resource](https://github.com/burdzwastaken/concourse-spinnaker-resource).
+
+## Source Configuration
+
+- `spinnaker_api`: the url of the Spinnaker api microservice. 
+- `spinnaker_application`: The Spinnaker application you would like to trigger.
+- `spinnaker_pipeline`: The Spinnaker pipeline you would like to trigger.
+- `skip_ssl_verification`: **TODO** make sure it is possible.
+- `client_x509_cert`: Client [certificate](https://www.spinnaker.io/setup/security/authentication/x509/) to authenticate with Spinnaker.
+- `client_x509_key`: Client [key](https://www.spinnaker.io/setup/security/authentication/x509/) to authenticate with Spinnaker.
+- `statuses`: *Optional* for the check step to filter the Spinnaker pipeline execution statuses.
+
+## Behaviour
+
+### `check`
+
+Pipeline executions will be found by fetching pipeline executions for the configured application, filtered by the pipeline name. If `statuses` is configured, the list will be filtered by statuses.
+
+The pipeline execution `id` will be used as the version of the resource.
+
+### `in`
+
+Places the following files in the destination:
+
+ - `metadata.json`: Contains the pipeline execution metadata returned from the Spinnaker [api](https://www.spinnaker.io/reference/api/docs.html#api-Pipelinecontroller-getPipelineUsingGET).
+
+ - `version`: A file containing the pipeline execution id.
+
+### `out`: Triggers a pipeline
+
+Triggers a Spinnaker pipeline.
```

Comment:
`out` behaviour may also want to check for specific results within the pipeline before moving to the next phase. an example of this is:https://github.com/novas0x2a/concourse-jenkins-resource#out-trigger-a-build